pkg/cmd/task/delete.go

```diff
@@ -75,11 +86,7 @@ tkn t rm foo -n bar
 	return c
 }
 
-func deleteTask(opts *deleteOptions, s *cli.Stream, p cli.Params, tName string) error {
-	cs, err := p.Clients()
-	if err != nil {
-		return fmt.Errorf("failed to create tekton client")
-	}
+func deleteTask(opts *deleteOptions, s *cli.Stream, p cli.Params, cs *cli.Clients, tName string) error {
```

Comment:
isn't cs already in `p cli.Params` ?

Reply:
Yes it is. I'll point on in a review comment why I went with this approach, but the gist of it was I was trying to avoid checking for the `"failed to create tekton client"` error twice when using `p.Clients()`.

Reply:
IMHO,  ... it is okay for internal (non exported methods) to make assumptions about arguments it receives ; i.e. the caller should have ensured the arguments are valid before passing it to callee so that certain step need to be performed only once ... (premature optimisation 🤷‍♂️ ; may be ...)